tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py

```diff
@@ -300,4 +300,47 @@ def test_1(self):
         z.backward()
 
 
+class TestConvolution2DInvalidDilation(unittest.TestCase):
+
+    n_batches = 2
+    in_channels = 3
+    out_channels = 2
+    dilate = 0
+    x_shape = (n_batches, in_channels, 10, 10)
+    w_shape = (out_channels, in_channels, 3, 3)
+
+    def check_invalid_dilation(self, x_data, w_data):
+        x = chainer.Variable(x_data)
+        w = chainer.Variable(w_data)
+        F.convolution_2d(x, w, dilate=self.dilate)
+
+    def test_invalid_dilation_cpu(self):
+        x = numpy.ones(self.x_shape, numpy.float32)
+        w = numpy.ones(self.w_shape, numpy.float32)
+        with self.assertRaises(ValueError):
+            with chainer.using_config('use_ideep', 'never'):
+                self.check_invalid_dilation(x, w)
+
+    def test_invalid_dilation_cpu_ideep(self):
+        x = numpy.ones(self.x_shape, numpy.float32)
+        w = numpy.ones(self.w_shape, numpy.float32)
+        with self.assertRaises(ValueError):
+            with chainer.using_config('use_ideep', 'always'):
+                self.check_invalid_dilation(x, w)
+
+    def test_invalid_dilation_gpu(self):
```

Comment:
Could you mark GPU tests?  Another option is to use `chainer.testing.inject_backend_tests`.

Reply:
Added `@gpu`, `@cudnn` and `@ideep` marks!

I suppose that `chainer.testing.inject_backend_tests` needs to be used with `chainer.testing.FunctionTestCase`, which does not support cases expecting an error raised as it is. Right?

Reply:
`@attr.gpu`, etc?

Reply:
Yes...